Enrich messages of SARIF results

Given the following code snippet:

```python
# test.py
import os

def greet():
    print('hello {name}'.format())
```

This is the default output:

```bash
$ ruff check test.py
test.py:1:8: F401 [*] `os` imported but unused
  |
1 | import os
  |        ^^ F401
2 | 
3 | def greet():
  |
  = help: Remove unused import: `os`

test.py:4:11: F524 `.format` call is missing argument(s) for placeholder(s): name
  |
3 | def greet():
4 |     print('hello {name}'.format())
  |           ^^^^^^^^^^^^^^^^^^^^^^^ F524
  |

Found 2 errors.
[*] 1 fixable with the `--fix` option.
```

If we try to get the SARIF output:

```bash
ruff check --output-format=sarif test.py
```
As shown below, the result messages (e.g. `UnusedImport` and `StringDotFormatMissingArguments`) are too short to be informative:

```json
{
  "$schema": "https://json.schemastore.org/sarif-2.1.0.json",
  "runs": [
    {
      "results": [
        {
          "level": "error",
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {...},
                "region": {
                  "endColumn": 10,
                  "endLine": 1,
                  "startColumn": 8,
                  "startLine": 1
                }
              }
            }
          ],
          "message": {
            "text": "UnusedImport"
          },
          "ruleId": "F401"
        },
        {
          "level": "error",
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {...},
                "region": {
                  "endColumn": 34,
                  "endLine": 4,
                  "startColumn": 11,
                  "startLine": 4
                }
              }
            }
          ],
          "message": {
            "text": "StringDotFormatMissingArguments"
          },
          "ruleId": "F524"
        }
      ],
      "tool": {...}
    }
  ],
  "version": "2.1.0"
}
```

Also per [the SARIF example](https://github.com/microsoft/sarif-tutorials/blob/main/docs/1-Introduction.md#a-simple-example), the message is as informative as Ruff's default (non-SARIF) one:

```json
"message": {
  "text": "'x' is assigned a value but never used."
}
```

--- a/crates/ruff_linter/src/message/sarif.rs
+++ b/crates/ruff_linter/src/message/sarif.rs
@@ -121,7 +121,7 @@ impl SarifResult {
         Ok(Self {
             rule: message.rule(),
             level: "error".to_string(),
-            message: message.name().to_string(),
+            message: message.body().to_string(),
             uri: url::Url::from_file_path(&path)
                 .map_err(|()| anyhow::anyhow!("Failed to convert path to URL: {}", path.display()))?
                 .to_string(),
@@ -215,6 +215,7 @@ mod tests {
             .unwrap();
         let results = sarif["runs"][0]["results"].as_array().unwrap();
         assert_eq!(results.len(), 3);
+        assert_eq!(results[0]["message"]["text"].as_str().unwrap(), "`os` imported but unused");
         assert!(rules.len() > 3);
     }
 }

--- a/crates/ruff_linter/src/message/sarif.rs
+++ b/crates/ruff_linter/src/message/sarif.rs
@@ -141,7 +141,7 @@ impl SarifResult {
         Ok(Self {
             rule: message.rule(),
             level: "error".to_string(),
-            message: message.name().to_string(),
+            message: message.body().to_string(),
             uri: path.display().to_string(),
             start_line: start_location.row,
             start_column: start_location.column,

--- a/crates/ruff_linter/src/message/sarif.rs
+++ b/crates/ruff_linter/src/message/sarif.rs
@@ -215,7 +215,10 @@ mod tests {
             .unwrap();
         let results = sarif["runs"][0]["results"].as_array().unwrap();
         assert_eq!(results.len(), 3);
-        assert_eq!(results[0]["message"]["text"].as_str().unwrap(), "`os` imported but unused");
+        assert_eq!(
+            results[0]["message"]["text"].as_str().unwrap(),
+            "`os` imported but unused"
+        );
         assert!(rules.len() > 3);
     }
 }

